command.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func timerWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) *time.Timer {
+	return time.AfterFunc(timeout, func() {
+		if cmd != nil && cmd.Process != nil {
+			log.Warningf("Command timed out, killing")
+			if err := cmd.Process.Kill(); err != nil {
+				log.Warningf("Error trying to kill process: %s", err)
+			}
+		}
+	})
```

Comment:
What is this supposed to be doing? Right now, it's never calling `cmd`. After `timeout`, it checks to see if cmd is running, and claiming it timed out if it's running (although nothing ran it). I think what you're actually looking for is:
```go
func runWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) {
    cmd.Start()
    doneCh := make(chan struct{})
    go func() {
        cmd.Wait()
        close(doneCh)
    }
    select {
    case <-doneCh:
    case <-time.After(timeout):
        log.Warningf("Command timed out, killing")
        if err := cmd.Process.Kill(); err != nil {
            log.Warningf("Error trying to kill process: %s", err)
        }
    }
}
```